Author: Agustin Martin Domingo <agmartin@debian.org>
Description: Work around wrong defface format for XEmacs

XEmacs only supports the old format in defface definitions and does not
accept extras to :underline.

To be integrated upstream. Will not show a reasonable overlay for
Emacs < 24.3, but this will go into a separate patch.

Index: dictionaries-common/support/emacsen/flyspell.el
===================================================================
--- dictionaries-common.orig/support/emacsen/flyspell.el	2014-10-14 18:27:33.327495656 +0200
+++ dictionaries-common/support/emacsen/flyspell.el	2014-10-14 18:27:33.323495665 +0200
@@ -447,19 +447,23 @@
 ;;*    Highlighting                                                     */
 ;;*---------------------------------------------------------------------*/
 (defface flyspell-incorrect
-  '((((supports :underline (:style wave)))
-     :underline (:style wave :color "Red1"))
-    (t
-     :underline t :inherit error))
+  (if (featurep 'xemacs)
+      '((((class color)) (:foreground "OrangeRed" :bold t :underline t))
+	(t (:bold t)))
+    '((((supports :underline (:style wave)))
+       :underline (:style wave :color "Red1"))
+      (t :underline t :inherit error)))
   "Flyspell face for misspelled words."
   :version "24.4"
   :group 'flyspell)
 
 (defface flyspell-duplicate
-  '((((supports :underline (:style wave)))
-     :underline (:style wave :color "DarkOrange"))
-    (t
-     :underline t :inherit warning))
+  (if (featurep 'xemacs)
+      '((((class color)) (:foreground "Gold3" :bold t :underline t))
+	(t (:bold t)))
+    '((((supports :underline (:style wave)))
+       :underline (:style wave :color "DarkOrange"))
+      (t :underline t :inherit warning)))
   "Flyspell face for words that appear twice in a row.
 See also `flyspell-duplicate-distance'."
   :version "24.4"
